snapshotter/devmapper/pool_device.go

```diff
@@ -73,87 +73,190 @@ func NewPoolDevice(ctx context.Context, config *Config) (*PoolDevice, error) {
 	}, nil
 }
 
+// transition invokes 'fn' callback to perform devmapper operation and reflects device state changes/errors in meta store.
+// 'tryingState' will be set before invoking callback. If callback succeeded 'successState' will be set, otherwise
+// error details will be recorded in meta store.
+func (p *PoolDevice) transition(ctx context.Context, deviceName string, tryingState DeviceState, successState DeviceState, fn func() error) error {
```

Comment:
Can we also have `fn` be named something a little more explicit since it is an ambiguous closure? Something like `devMapperFn func() error`

Reply:
would `transitionFn` work for you ?